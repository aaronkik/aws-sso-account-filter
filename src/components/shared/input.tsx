import { ComponentProps, forwardRef } from 'react';
import { twMerge } from 'tailwind-merge';

const Input = forwardRef<HTMLInputElement, ComponentProps<'input'>>(
  ({ className, ...props }, ref) => (
    <input
      className={twMerge(
        'w-full rounded-sm border border-transparent bg-neutral-200 leading-6 placeholder:text-neutral-400 focus:border-orange-400 focus:bg-transparent focus:ring-orange-400',
        className
      )}
      ref={ref}
      {...props}
    />
  )
);

Input.displayName = 'Input';

export default Input;
