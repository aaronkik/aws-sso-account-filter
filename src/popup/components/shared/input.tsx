import { type ComponentPropsWithRef, forwardRef } from 'react';
import { twMerge } from 'tailwind-merge';

const Input = forwardRef<HTMLInputElement, ComponentPropsWithRef<'input'>>(({ className, ...props }, ref) => (
  <input
    className={twMerge(
      'w-full rounded-sm border border-transparent bg-slate-700 leading-6 placeholder:text-slate-500 focus:border-orange-400 focus:bg-transparent focus:ring-orange-400 focus:placeholder:text-slate-600',
      className,
    )}
    ref={ref}
    {...props}
  />
));

Input.displayName = 'Input';

export default Input;
