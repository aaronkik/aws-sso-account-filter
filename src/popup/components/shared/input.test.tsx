import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import { Input } from './input';

describe('Input', () => {
  test('matches snapshot', () => {
    const { container } = render(<Input />);
    expect(container).toMatchSnapshot();
  });
});
