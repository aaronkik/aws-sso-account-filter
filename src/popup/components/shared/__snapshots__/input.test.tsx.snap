// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > matches snapshot 1`] = `
<div>
  <input
    class="w-full rounded-sm border border-transparent bg-slate-800 leading-6 placeholder:text-slate-500 focus:border-orange-400 focus:bg-transparent focus:ring-orange-400 focus:placeholder:text-slate-400"
  />
</div>
`;
